packages/paste-core/components/datagrid/src/DataGrid.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import {Box} from '@twilio-paste/box';
+import type {BoxElementProps} from '@twilio-paste/box';
+import {useUID} from '@twilio-paste/uid-library';
+import {useCompositeState, Composite} from '@twilio-paste/reakit-library';
+import {Table} from '@twilio-paste/table';
+import type {TableProps} from '@twilio-paste/table';
+import {DataGridContext} from './DataGridContext';
+
+import {
+  ensureFocus,
+  getActiveElement,
+  getFirstFocusableIn,
+  getClosestCellFrom,
+  getClosestGridCellFromCurrentFocus,
+  isCell,
+  delayedSetFocusable,
+} from './utils';
+
+interface DataGridProps extends TableProps {
+  'aria-label': string;
+  element?: BoxElementProps['element'];
+}
+
+export const DataGrid = React.forwardRef<HTMLTableElement, DataGridProps>(({element = 'DATAGRID', ...props}, ref) => {
+  const dataGridId = `datagrid-${useUID()}`;
+  const lastFocusedElement = React.useRef<Element | null>(null);
+  const compositeState = useCompositeState({unstable_virtual: false});
+  const [actionable, setActionable] = React.useState<boolean>(false);
+
+  const handleMouseDown = React.useCallback(() => {
+    if (!actionable) {
+      setActionable(true);
+    }
+  }, [actionable]);
+
+  /**
+   * Keep track of the last focused element. This is needed to restore tabIndex to 0
+   * when clicking out of the grid, so that we can tab back into it.
+   */
+  const handleFocus = React.useCallback((e: React.FocusEvent) => {
+    if (e.target != null) {
+      lastFocusedElement.current = e.target;
+    }
+  }, []);
+
+  /**
+   * - Reset dataGrid to navigational on blur
+   * - Sets the last focused element before blurring to be the active tab stop (line 43)
+   */
+  const handleBlur = React.useCallback(
+    (event) => {
+      const isDataGridBlurred = !event.currentTarget.contains(event.relatedTarget);
+
+      if (isDataGridBlurred && actionable) {
+        setActionable(false);
```

Comment:
This one should use the functional update.

Reply:
I need to unconditionally set this to false, I'm not using the previous state to infer the new state so it can remain as it is.